libs/core/datastructures/include/hpx/datastructures/detail/variant.hpp

```diff
@@ -0,0 +1,2860 @@
+// MPark.Variant
+//
+// Copyright Michael Park, 2015-2017
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+
+#ifndef MPARK_VARIANT_HPP
+#define MPARK_VARIANT_HPP
```

Comment:
`#pragma once`? But only if we change the `mpark` namespace below to `hpx`...

Reply:
Yep, no worries. Not having had changes in a long time is an argument in both directions ;) On one hand it means we hopefully won't need to update it often and can make changes, and on the other hand it also means that if someone else is using it they most likely have the same version. My concern is the same as with the moodycamel queue, i.e. that if someone else has a slightly different version of the same header things might get weird, but I don't mind the easy route of keeping it as it is until someone has a problem (which would anyway only be on Windows).

Reply:
If we added `#pragma once`, we would have to change the namespace as well. I can do both.